aspnetcore/blazor/fundamentals/routing.md

```diff
@@ -33,77 +44,96 @@ Interactive routing also prevents prerendering because new page content isn't re
+:::moniker-end
 
-At runtime, the <xref:Microsoft.AspNetCore.Components.RouteView> component:
+:::moniker range="< aspnetcore-6.0"
 
-* Receives the <xref:Microsoft.AspNetCore.Components.RouteData> from the <xref:Microsoft.AspNetCore.Components.Routing.Router> along with any route parameters.
-* Renders the specified component with its [layout](xref:blazor/components/layouts), including any further nested layouts.
+Blazor Server is integrated into [ASP.NET Core Endpoint Routing](xref:fundamentals/routing). An ASP.NET Core app is configured to accept incoming connections for interactive components with <xref:Microsoft.AspNetCore.Builder.ComponentEndpointRouteBuilderExtensions.MapBlazorHub%2A> in `Startup.Configure`.
 
-Optionally specify a <xref:Microsoft.AspNetCore.Components.RouteView.DefaultLayout> parameter with a layout class for components that don't specify a layout with the [`@layout` directive](xref:blazor/components/layouts#apply-a-layout-to-a-component). The framework's [Blazor project templates](xref:blazor/project-structure) specify the `MainLayout` component (`MainLayout.razor`) as the app's default layout. For more information on layouts, see <xref:blazor/components/layouts>.
+:::moniker-end
 
-Components support multiple route templates using multiple [`@page` directives](xref:mvc/views/razor#page). The following example component loads on requests for `/blazor-route` and `/different-blazor-route`.
+:::moniker range="< aspnetcore-8.0"
 
-`BlazorRoute.razor`:
+The typical configuration is to route all requests to a Razor page, which acts as the host for the server-side part of the Blazor Server app. By convention, the *host* page is usually named `_Host.cshtml` in the `Pages` folder of the app.
 
-:::moniker range=">= aspnetcore-9.0"
+The route specified in the host file is called a *fallback route* because it operates with a low priority in route matching. The fallback route is used when other routes don't match. This allows the app to use other controllers and pages without interfering with component routing in the Blazor Server app.
 
-:::code language="razor" source="~/../blazor-samples/9.0/BlazorSample_BlazorWebApp/Components/Pages/BlazorRoute.razor":::
+For information on configuring <xref:Microsoft.AspNetCore.Builder.RazorPagesEndpointRouteBuilderExtensions.MapFallbackToPage%2A> for non-root URL server hosting, see <xref:blazor/host-and-deploy/app-base-path>.
 
 :::moniker-end
 
-:::moniker range=">= aspnetcore-8.0 < aspnetcore-9.0"
+## Route templates
+
+:::moniker range=">= aspnetcore-8.0"
 
-:::code language="razor" source="~/../blazor-samples/8.0/BlazorSample_BlazorWebApp/Components/Pages/BlazorRoute.razor":::
+The <xref:Microsoft.AspNetCore.Components.Routing.Router> component enables routing to Razor components and is located in the app's `Routes` component (`Components/Routes.razor`).
```

Comment:
This mention of the `Router` component feels a bit out of place to me. The routing doc really needs to cover two things: 1. How to add routes to components (`@page`, route templates, route constraints, route parameters), and 2. How to set up routing in your app (`MapComponents`, `Router`, static vs interactive routing, etc.). The `Router` component is important enough that I think it probably deserves its own section. We don't have to deal with that as part of this PR though.